src/dsl/schema/deserialization.rs

```diff
@@ -18,19 +18,25 @@ where
     E: serde::de::Error,
 {
     let maybe_root = schema.as_mapping();
+    const DEFAULT_VERSION: u64 = 1;
     let version = match maybe_root {
         Some(mapping) => Ok({
-            let version = mapping
-                .get(&Value::from("version"))
-                .ok_or_else(|| CompilationError::with_message("you must specify schema version"))?;
-            version
-                .as_u64()
-                .ok_or_else(|| CompilationError::with_message("version must be a positive integer"))?
+            let version = mapping.get(&Value::from("version"));
+
+            match version {
+                Some(version) => Some(
+                    version
+                        .as_u64()
+                        .ok_or_else(|| CompilationError::with_message("version must be a positive integer"))?,
```

Comment:
Since we're touching this ...

* version must be a positive integer
* and if provided, it must equal to 1 otherwise it's invalid

Reply:
alrighty :)

Reply:
Added a unit test for both cases and the check